File: frontend/src/dto/auction/addAuctionDto.js
import Joi from "joi";

class AddAuctionDto{

    auctionStartTime = "";
    auctionEndTime = "";
}

export function validateAddAuctionDto(addAuctionDto){

    const schema = Joi.object({
     
        auctionStartTime: Joi.date()
            .min('now')
            .required()
            .messages({
                'date.min': 'Auction start time must be today or in the future.',
            }),
        auctionEndTime: Joi.date()
            .greater(Joi.ref('auctionStartTime'))
            .required()
            .custom((value, helpers) => {
                const auctionStartTime = helpers.state.ancestors[0].auctionStartTime;
                if (value < new Date(new Date(auctionStartTime).getTime() + 10 * 60000)) {
                    return helpers.message('Auction end time must be at least 10 minutes after the start time.');
                }
                return value;
            })
            .messages({
                'date.greater': 'Auction end time must be after the start time.',
            }),
    });

    return schema.validate(addAuctionDto, { abortEarly: false });
}


export default AddAuctionDto;

File: backend/DTO/Auction/AddAuctionDto.cs
using System.ComponentModel.DataAnnotations;

namespace DreamBid.Dtos.Auction
{
    public class AddAuctionDto : IValidatableObject
    {

        private DateTime? _AuctionStartTime { get; set; }
        [Required]
        public DateTime? AuctionStartTime { get => _AuctionStartTime != null ? DateTime.SpecifyKind((DateTime)_AuctionStartTime, DateTimeKind.Utc) : null; set => _AuctionStartTime = DateTime.SpecifyKind((DateTime)value, DateTimeKind.Utc); }

        private DateTime? _AuctionEndTime { get; set; }

        [Required]
        public DateTime? AuctionEndTime { get => _AuctionEndTime != null ? DateTime.SpecifyKind((DateTime)_AuctionEndTime, DateTimeKind.Utc) : null; set => _AuctionEndTime = DateTime.SpecifyKind((DateTime)value, DateTimeKind.Utc); }


        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            if (AuctionStartTime < DateTime.UtcNow)
            {
                yield return new ValidationResult("Auction start time must be today or in the future.");
            }

            if (AuctionEndTime < AuctionStartTime?.AddMinutes(10))
            {
                yield return new ValidationResult("Auction end time must be at least 10 minutes after the start time.");
            }

            yield return ValidationResult.Success;
        }
    }
}